File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq_dict = {chr(i): 0 for i in range(ord('A'), ord('Z') + 1)}
        left, right = 0, 0
        max_freq = 0
        max_len = 0
        while right < len(s):
            freq_dict[s[right]] += 1
            max_freq = max(max_freq, freq_dict[s[right]])
            if (right - left + 1) - max_freq > k:
                freq_dict[s[left]] -= 1
                left += 1
            max_len = max(max_len, right - left + 1)
            right += 1
        return max_len

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
            int mx=0;
        int n=s.length();
        for(char c='A';c<='Z';c++){
            int start=0, i=0, cnt=0;
            for(i=0;i<n;i++){
                if(s[i]!=c){
                    cnt++;   
                }
                while(cnt>k){
                    mx=max(mx,i-start);
                    if(s[start]!=c)cnt--;
                    start++;
                }
            }
            mx=max(mx,i-start);
        }
        return mx;
    
    }
};